client/packages/sprotty-client/src/features/tool-feedback/change-bounds-tool-feedback.ts

```diff
@@ -170,18 +299,64 @@ export class FeedbackMoveMouseListener extends MouseListener {
                                 mouseOverElement = true;
                                 result.push(new ApplyCursorCSSFeedbackAction(CursorCSS.DEFAULT));
                             }
+
+                            const selectedElements: FluentIterable<SModelElement> = target.root.index.all()
+                                .filter(selected => isSelectable(selected) && selected.selected);
+
+                            // If the ghost element has moved beyond the obstacle move the actual element there aswell
+                            // But only if a single element is selected (multi-selection jumps are not supported)
+                            if (this.hasCollided && collisionTargetsGhost.length === 0 && toArray(selectedElements).length === 1) {
+                                mouseOverElement = true;
+                                result.push(new ApplyCursorCSSFeedbackAction(CursorCSS.DEFAULT));
+
+                                if (element.id === ghostElement.id) {
+                                    element.bounds = ghostElement.bounds;
+                                }
+
+                            }
                             // Get only the valid, non-slected collision targets to avoid in-selection collisions
                             const collisionTargets: SModelElement[] = this.getCollisionChain(target, element, dx, dy, [])
                                 .filter(collidingElement => isSelectable(collidingElement) && !collidingElement.selected);
 
                             if (collisionTargets.length > 0) {
                                 collisionTargets.forEach(collisionTarget => {
                                     if (isResizeable(collisionTarget)) {
+                                        // Only snap on first collision to avoid erratic jumps
+                                        if (!this.hasCollided) {
+                                            const snappedBounds = this.getSnappedBounds(element, collisionTarget);
+                                            const snapMoves: ElementMove[] = [];
+                                            snapMoves.push({
+                                                elementId: element.id,
+                                                fromPosition: {
+                                                    x: element.position.x,
+                                                    y: element.position.y
+                                                },
+                                                toPosition: {
+                                                    x: snappedBounds.x,
+                                                    y: snappedBounds.y
+                                                }
+                                            });
+                                            result.push(new MoveAction(snapMoves, false));
+                                        }
+
+
```

Comment:
Sorry, github doesn't allow me to comment on code that hasn't changed. So this comment is about the lines below.
In line 368 and 369, there is a very complex condition that is hard to follow.
Also, is it really necessary to access `willCollide` and `hasCollided` down there, in a block that is not inside the code for the case where `noElementOverlap` is on? I find it a bit confusing to leak those variables into the "I don't care about collisions" case.
If a state has to leak, the switch should be very general, e.g. `isMoveAllowed`.
Lastly, couldn't we make the terminology a bit more consistent wrt element overlap and collision. We have the switch `noElementOverlap`, why don't we name the `willCollide` also `willOverlap`?